File: Array/018-Intersection_Arrays/Intersection_Arrays.js
// Clear & straightforward solution

var intersection = function(nums1, nums2) {
    let out = []
    for (let i=0; i<nums1.length; i++) {
        if (nums2.includes(nums1[i])) {
           out.push(nums1[i])
        }
    }
    
   let setOut = Array.from(new Set(out))
   return setOut
};

// Simplest solution

var intersection = function(nums1, nums2) {
    return [...new Set(nums1)].filter(num => nums2.includes(num));
};

File: Array/018-Intersection_Arrays/Intersection_Arrays.go
func intersection(nums1 []int, nums2 []int) []int {
    var numMap = map[int]bool{}
    var numArr = []int{}
    
    for _,num := range nums1 {
        numMap[num] = true
    }
    
    for _,num := range nums2 {
        if numMap[num] == true {
            numArr = append(numArr, num)
            numMap[num] = false
        }
    }
    
    return numArr
}
